packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -105,6 +105,11 @@ class FlatTermSelector extends Component {
 
 	componentDidUpdate( prevProps ) {
 		if ( prevProps.terms !== this.props.terms ) {
+			invoke( this.initRequest, [ 'abort' ] );
+			this.initRequest = this.fetchTerms( {
+				include: this.props.terms.join( ',' ),
+				per_page: -1,
+			} );
```

Comment:
This feels like a good fix. The downside is that it will probably trigger a useless API request when you add new tags using the form (instead of the data module). A potential way to avoid this is to check whether all the terms in `this.props.terms` are already loaded in the "availableTerms"

Reply:
Indeed, it would be more efficient. After, this bug is reported as part of an open-source plugin, which loads tags elsewhere, so the API call implies that the terms will not be loaded.